### PR TITLE
First step towards CD. (Only flair for now)

### PR DESCRIPTION
**Pull request by Narsil**
The goal is to automatically deploy on merge.
The helm prod chart lives in another repo, ideally we trigger
another action in this other repo (it's simpler). 

The building and uploading can happen here, we just need to add some secrets to this repo.

Feedback welcome on the approach.

Ideally I would like to have a single CD action, that would just check at the modified folders and deploy every folder hit in `docker_images`.

This seems doable however not highly stable (depends on the trigger of the action to get the correct git commands)

Requires: https://github.com/huggingface/api-inference/pull/1151

### CODE REVIEW

File: .github/workflows/python-api-flair-cd.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Install dependencies
```
Comment (OlivierDehaene) [inline .github/workflows/python-api-flair-cd.yaml:22]: I don't think we need kubectl, helm kubernetes and uvicorn(?)

File: .github/workflows/python-api-flair-cd.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        run: |
+          pip install --upgrade pip
+          pip install kubernetes httpx uvicorn awscli
+          sudo snap install kubectl --classic
+          helm plugin install https://github.com/databus23/helm-diff
+      - name: Setup K8S
```
Comment (OlivierDehaene) [inline .github/workflows/python-api-flair-cd.yaml:28]: Should not be needed.

**Review by osanseviero — COMMENTED** [review]
Overall looks like a good approach, thanks a lot for working on this!

File: .github/workflows/python-api-flair-cd.yaml (hunk 1 above)
Comment (Narsil) [reply .github/workflows/python-api-flair-cd.yaml:22]: Thanks for the catch !

File: .github/workflows/python-api-flair-cd.yaml (hunk 2 above)
Comment (Narsil) [reply .github/workflows/python-api-flair-cd.yaml:28]: Thanks for the catch !

**Review by OlivierDehaene — APPROVED** [review]

**Pull request merged**